app/pages/lab/media-area/index.jsx

```diff
@@ -35,25 +38,43 @@ export default class MediaAreaController extends React.Component {
 
   componentWillReceiveProps(nextProps) {
     if (this.props.resource !== nextProps.resource) {
-      this.fetchMedia(nextProps);
+      this.fetchMedia(nextProps, 1);
     }
   }
+  
+  onPageChange (page) {
+    // When user requests the page to change, the media fetch AND the update
+    // of state.page is done in fetchMedia()
+    this.fetchMedia(this.props, page);
```

Comment:
I think the code here should be something like this, to keep component state inside the component:
```js
this.setState({ page });
this.fetchMedia(this.props, page)
  .then(({ media, page_count }) => {
    this.setState({ media, page_count });
  });
```
I've assumed that `fetchMedia` resolves errors, which may or may not be great for UX.
```